sycl/include/sycl/ext/oneapi/experimental/enqueue_functions.hpp

```diff
@@ -183,11 +193,15 @@ template <int Dimensions, typename Properties, typename... ArgsT>
 void parallel_for(handler &CGH,
                   launch_config<range<Dimensions>, Properties> Config,
                   const kernel &KernelObj, ArgsT &&...Args) {
+  static_assert(detail::NoPropertyHasCompileTimeKernelEffect<Properties>::value,
+                "This kernel enqueue function does not allow properties with "
```

Comment:
Kind of a nit, but I'd consider rephrasing this as:
```suggestion
                "launch_config does not allow properties with "
```
It's still possible to use properties that have compile-time kernel effects, but they have to be attached to the kernel definition instead.

Actually... Don't we need a static assert like this on the overloads accepting a lambda/functor as well as the overloads accepting a `sycl::kernel`?

Reply:
The assert has been moved to `launch_config` instead.